### ISSUE TITLE
Hook jank into leiningen

### ISSUE THREAD

**Issue by jeaye** [jank-lang/jank#29]
I'm not sure how this will work yet, if jank will have a lein plugin or if something else is required. But jank will be adding support for `require` shortly, so we should be able to start hooking in deps, generating compilation artifacts, etc.

For now, blocked on `require` support.

**Comment by Samy-33** [comment]
@jeaye, please assign this to me

**Comment by amano-kenji** [comment]
leiningen requires JVM. Would it not be bad for jank to require JVM? Jank is going to be hosted on C++ and LLVM. It should have its own toolchain independent from JVM?

**Comment by jeaye** [comment]
> leiningen requires JVM. Would it not be bad for jank to require JVM? Jank is going to be hosted on C++ and LLVM. It should have its own toolchain independent from JVM?

We'll start by ensuring we have lein support. This provides a good migration path for anyone using Clojure JVM. jank may develop a replacement in the future. We may also work with technomancy to make a jank version of leiningen. I'm not sure yet. For now, especially since jank hasn't hit Clojure parity and we have far more pressing matters, leiningen *works*. That's important.

**Comment by amano-kenji** [comment]
If you start with lein support, make sure leiningen behaves as a system/project/venv package manager like python's pip and janet's jpm. Otherwise, I wouldn't be able to use jank as a system scripting language whose libraries can be installed as linux packages or venv/local-project dependencies.

Python finally got this right with pyproject.toml.

**Comment by alysbrooks** [comment]
Another option would be to have Leiningen binaries compiled using Graalvm native. I'm honestly not sure whether that would be easier or harder than making a Jank version. In theory it should be easier—GraalVM trying to maximize compatibility with existing code and all—but I imagine a project manager would rely on dynamic execution in ways a typical JVM project wouldn't.

**Comment by jeaye** [comment]
> Another option would be to have Leiningen binaries compiled using Graalvm native.

Yep, I've spoken with technomancy about this and it's a possible way forward. For now, the experience we have is still a big step up from Clojure CLI.

**Comment by jeaye** [comment]
This is done, thanks to Kyle's work.
